### DISCUSSION TITLE
How to add 'referrerpolicy' for image or link when render?

### DISCUSSION THREAD

**Discussion by 1s1y** [remarkjs/remark#1102] [Q&A] [answered]
Hello!
Is there any way to render `referrerpolicy` with image or link element?
Modify DOM attribute in `useEffect` may be a workaround. But I'm not sure `useEffect` will be invoked before image request.

**Comment by wooorm** [comment, marked as answer]
a) don‘t do it in remark, remark is about markdown, this is about HTML, so you do it in rehype
b) [traverse the tree](https://unifiedjs.com/learn/recipe/tree-traversal/), looking for elements sort of [like this](https://github.com/rehypejs/rehype-external-links/blob/main/index.js), and set the `referrerPolicy` property to `true`
c) there are tons of ways to use remark, there will be different answers, depending on what other tools you use

**Reply by 1s1y** [reply]
good idea! It seems not difficult to make a similar rehype plugin. I'll try it.
